R/capacity/disasters-emergencies/england/la-spending-power.R

```diff
@@ -0,0 +1,228 @@
+# Combined Authority lookup source: https://geoportal.statistics.gov.uk/datasets/ons::local-authority-district-to-combined-authority-december-2020-lookup-in-england/about
+response <- httr::GET("https://services1.arcgis.com/ESMARspQHYMw9BZ9/arcgis/rest/services/LAD20_CAUTH20_EN_LU/FeatureServer/0/query?where=1%3D1&outFields=*&outSR=4326&f=json")
+content <- httr::content(response, type = "application/json", simplifyVector = TRUE)
+
+combined_auth_lookup <- content$features$attributes |>
+  select(-column4, 
+         -FID)
+
+# There are no changes between 2020 and 2021 LAD codes for Greater Manchester LTLAs
+combined_auth_ltla_cps <- remaining_utla |>
+  inner_join(combined_auth_lookup, by = c("lad_code" = "CAUTH20CD")) |>
+  left_join(lad_pop_update, by = c("LAD20CD" = "LAD21CD")) |>
+  mutate(pop_weighted_cps = pop / sum(pop) * cps_millions) |>
+  select(LAD21CD = LAD20CD, 
+         cps_millions)
+
+# Check if these LTLAs from splitting UTLAs are already in the data as LTLAs
+combined_auth_ltla_cps |>
+  inner_join(ltla_cps, by = "LAD21CD")
+# Yes they are already in the LTLA level data so this spending power is in addition
+# to the LTLA level spending power
+
+# Greater London Authority ----
+# Source: https://data.london.gov.uk/dataset/london-borough-profiles
+# Issues with column names having special characters
+gla_clean <- read_csv("https://data.london.gov.uk/download/london-borough-profiles/c1693b82-68b1-44ee-beb2-3decf17dc1f8/london-borough-profiles.csv",
+  col_names = FALSE
+)
+
+# Cleaning and removing any cumulative areas that are just combinations of the LTLAs
+gla_lad_codes <- gla_clean |>
+  select(lad_code = X1, 
+         area = X2, 
+         inner_outer = X3) |>
+  slice(-1) |>
+  filter(!is.na(inner_outer))
+
+gla_lad_codes |>
+  anti_join(lad_pop_update, by = c("lad_code" = "LAD21CD"))
+
+gla_cps_value <- remaining_utla |>
+  filter(lad_name == "Greater London Authority") |>
+  pull(cps_millions)
+
+gla_ltla_cps <- gla_lad_codes |>
+  mutate(cps_millions = gla_cps_value) |>
+  left_join(lad_pop_update, by = c("lad_code" = "LAD21CD")) |>
+  mutate(pop_weighted_cps = pop / sum(pop) * cps_millions) |>
+  select(LAD21CD = lad_code, 
+         cps_millions = pop_weighted_cps)
+
+# Check if these LTLAs from splitting GLA are already in the data as LTLAs
+gla_ltla_cps |>
+  inner_join(ltla_cps, by = "LAD21CD")
+# Yes they are already in the LTLA level data so this spending power is in addition
+# to the LTLA level spending power
+
+# Combining all the data sets ---
+combined <- ltla_cps |>
+  bind_rows(ltla_fire_cps) |>
+  bind_rows(ltla_utla_cps_updated) |>
+  bind_rows(combined_auth_ltla_cps) |>
+  bind_rows(gla_ltla_cps) |>
+  group_by(lad_code = LAD21CD) |>
+  summarise(cps_millions = sum(cps_millions))
+
+# Save data ----
+combined |>
+  write_rds("data/capacity/disasters-emergencies/england/la-spending-power.rds")
+
+# To email about the dataset to find out why some UTLAs get additional spending power (and not others)
```

Comment:
I am interested to find out why some LTLA's get additional spending power in the form of combined authorities, etc. Please keep us posted when you hear more. My worry is that the same funds are being double counted each time the LAD codes repeat.

Reply:
I did get an initial response on this noted https://github.com/britishredcrosssociety/resilience-index/blob/f260c40d0f5fd380e4d7a2ab258a8f787e1bf3d3/R/capacity/disasters-emergencies/england/la-spending-power.R#L123 so I don't think it's double counted and more split between UTLA and LTLA. Will feedback when get further reply though.